drivers/audio/CMakeLists.txt

```diff
@@ -7,3 +7,4 @@ zephyr_library_sources_ifdef(CONFIG_AUDIO_MPXXDTYY	mpxxdtyy.c)
 zephyr_library_sources_ifdef(CONFIG_AUDIO_MPXXDTYY	mpxxdtyy-i2s.c)
```

Comment:
Can you please clarify what is an 'audio codec shell command'? Does Zephyr has a 'shell' like normal Linux has? Also if this is so, why do you place your code inside `drivers/audio`  where for what i have seen there are only audio codec implementations.

Reply:
> Does Zephyr has a 'shell' like normal Linux has?

Yes, Zephyr has a shell (`CONFIG_SHELL` + [official doc](https://docs.zephyrproject.org/latest/services/shell/index.html)).
It is super conveniant for test and debug.
For example with `audio` command, my hardware team colleague was able to measure board power consumption at different volumes easily.

> Also if this is so, why do you place your code inside `drivers/audio` where for what i have seen there are only audio codec implementations.

All other driver related shell commands are in driver folder:

```
drivers/adc/adc_shell.c
drivers/bbram/bbram_shell.c
drivers/can/can_shell.c
drivers/counter/counter_timer_shell.c
drivers/dac/dac_shell.c
drivers/edac/shell.c
drivers/eeprom/eeprom_shell.c
drivers/flash/flash_shell.c
drivers/fpga/fpga_shell.c
drivers/gpio/gpio_shell.c
drivers/hwinfo/hwinfo_shell.c
drivers/i2c/i2c_shell.c
drivers/led/led_shell.c
drivers/lora/shell.c
drivers/mdio/mdio_shell.c
drivers/modem/modem_shell.c
drivers/pcie/host/shell.c
drivers/pm_cpu_ops/psci_shell.c
drivers/pwm/pwm_shell.c
drivers/regulator/regulator_shell.c
drivers/rtc/rtc_shell.c
drivers/sensor/sensor_shell.c
drivers/sensor/shell_battery.c
drivers/smbus/smbus_shell.c
drivers/virtualization/virt_ivshmem_shell.c
drivers/w1/w1_shell.c
drivers/watchdog/wdt_shell.c
drivers/wifi/eswifi/eswifi_shell.c
```